experimental/Schemes/WeilDivisor.jl

```diff
@@ -210,34 +210,49 @@ function intersect(D::WeilDivisor, E::WeilDivisor)
   return result
 end
 
-function colength(I::IdealSheaf)
+function colength(I::IdealSheaf; covering::Covering=default_covering(scheme(I)))
   X = scheme(I)
-  C = default_covering(X)
   patches_todo = copy(affine_charts(X))
   patches_done = AbsSpec[]
   result = 0
   while length(patches_todo) != 0
     U = pop!(patches_todo)
     J = I(U)
-    # To avoid overcounting, throw away all components that 
-    # were already visible in other charts.
-    for V in patches_done
-      if !haskey(glueings(C), (U, V))
-        continue
+    if has_decomposition_info(covering)
```

Comment:
Before doing any further computations (i.e. above this line), you might want to check that the colength is finite -- to avoid being stuck in a loop forever.

Reply:
But this is another groebner basis computation and therefore should be hidden behind an `@check`.

Reply:
Please add this input check.